src/ui/steps/AddStep.cpp

```diff
@@ -4,10 +4,9 @@
 
 #include "AddStep.h"
 
-AddStep::AddStep(std::shared_ptr<AbstractReader> reader,
-                 std::shared_ptr<AbstractPrinter> printer,
-                 std::shared_ptr<Machine> submachine) :
-         Step(reader, printer) {
+AddStep::AddStep(const std::shared_ptr<AbstractPrinter> &printer,
+                 const std::shared_ptr<Machine> &submachine) :
+         PrinterStep(printer) {
     submachine_ = submachine;
```

Comment:
Why not initialize it in the initializer list?

Reply:
updated